### PR TITLE
MULE-9132: Ensure proper closing of Statements and ResultSets

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/db/src/main/java/org/mule/module/db/internal/result/statement/EagerStatementResultHandler.java [hunk 1]
```diff
@@ -29,16 +31,28 @@ public EagerStatementResultHandler(ResultSetHandler resultSetHandler)
...
 
     @Override
-    public Object processStatement(DbConnection connection, Statement statement, QueryTemplate queryTemplate, AutoGeneratedKeyStrategy autoGeneratedKeyStrategy) throws SQLException
+    public Map<String, Object> processStatement(DbConnection connection, Statement statement, QueryTemplate queryTemplate, AutoGeneratedKeyStrategy autoGeneratedKeyStrategy) throws SQLException
     {
+        Map<String, Object> resultsMap = null;
         try
         {
-            return super.processStatement(connection, statement, queryTemplate, autoGeneratedKeyStrategy);
+            resultsMap = super.processStatement(connection, statement, queryTemplate, autoGeneratedKeyStrategy);
```
Comment (pablokraan) [inline modules/db/src/main/java/org/mule/module/db/internal/result/statement/EagerStatementResultHandler.java]: Revert this change

File: modules/db/src/main/java/org/mule/module/db/internal/result/statement/EagerStatementResultHandler.java [hunk 2]
```diff
@@ -29,16 +31,28 @@ public EagerStatementResultHandler(ResultSetHandler resultSetHandler)
...
         }
         finally
         {
             try
             {
+                if (resultsMap != null)
+                {
+                    for (Object resultValue : resultsMap.values())
+                    {
+                        if (resultValue instanceof ResultSet)
```
Comment (pablokraan) [inline modules/db/src/main/java/org/mule/module/db/internal/result/statement/EagerStatementResultHandler.java]: Is this code tested?

File: modules/db/src/test/java/org/mule/module/db/integration/select/SelectsLongTransactionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Override
+    protected String[] getFlowConfigurationResources()
+    {
+        return new String[] {"integration/select/selects-long-transaction-config.xml"};
+    }
+
+    @Test
+    public void oneWay() throws Exception
```
Comment (pablokraan) [inline modules/db/src/test/java/org/mule/module/db/integration/select/SelectsLongTransactionTestCase.java]: Is not required to test both one-way and request response scenarios

File: modules/db/src/test/java/org/mule/module/db/integration/select/SelectsLongTransactionTestCase.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+    public SelectsLongTransactionTestCase(String dataSourceConfigResource, AbstractTestDatabase testDatabase)
+    {
+        super(dataSourceConfigResource, testDatabase);
+    }
+
+    @Parameterized.Parameters
+    public static List<Object[]> parameters()
+    {
+        return TestDbConfig.getResources();
```
Comment (pablokraan) [inline modules/db/src/test/java/org/mule/module/db/integration/select/SelectsLongTransactionTestCase.java]: Do this test apply for all the different datasources or just Oracle?

File: modules/db/src/test/resources/integration/config/oracle-db-config.xml [hunk 5]
```diff
@@ -5,6 +5,6 @@
       xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
             http://www.mulesoft.org/schema/mule/db http://www.mulesoft.org/schema/mule/db/current/mule-db.xsd">
 
-    <db:oracle-config name="dbConfig" host="172.16.28.128" port="1521" instance="XE" user="pocholo"
-                      password="password"/>
+    <db:oracle-config name="dbConfig" host="localhost" port="1521" instance="cdb1" user="system"
```
Comment (pablokraan) [inline modules/db/src/test/resources/integration/config/oracle-db-config.xml]: Revert this change

File: modules/db/src/test/java/org/mule/module/db/integration/select/SelectsLongTransactionTestCase.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    public void requestResponse() throws Exception
+    {
+        LocalMuleClient client = muleContext.getClient();
+
+        MuleMessage response = client.send("vm://testRequestResponse", TEST_MESSAGE, null);
+
+        assertThat(response.getExceptionPayload(), nullValue());
+    }
+
+    public static class SetCollectionFlowVar implements Callable
```
Comment (pablokraan) [inline modules/db/src/test/java/org/mule/module/db/integration/select/SelectsLongTransactionTestCase.java]: Instead of using a component, you can send the collection as the message's payload in the test message

File: modules/db/src/main/java/org/mule/module/db/internal/result/statement/EagerStatementResultHandler.java (hunk 2 above)
Comment (elrodro83) [inline modules/db/src/main/java/org/mule/module/db/internal/result/statement/EagerStatementResultHandler.java]: I removed taht. Didn't realize that being eager there are no ResultSet's, only Map's
